src/thunderbots/software/ai/primitive/directwheels_primitive.cpp

```diff
@@ -0,0 +1,77 @@
+#include "ai/primitive/directwheels_primitive.h"
+
+const std::string DirectWheelsPrimitive::PRIMITIVE_NAME = "DirectWheels Primitive";
+
+DirectWheelsPrimitive::DirectWheelsPrimitive(unsigned int robot_id, int wheel0_power,
+                                             int wheel1_power, int wheel2_power,
+                                             int wheel3_power, double dribbler_rpm)
+    : robot_id(robot_id),
+      wheel0_power(wheel0_power),
+      wheel1_power(wheel1_power),
+      wheel2_power(wheel2_power),
+      wheel3_power(wheel3_power),
+      dribbler_rpm(dribbler_rpm)
+{
+}
+
+DirectWheelsPrimitive::DirectWheelsPrimitive(
+    const thunderbots_msgs::Primitive &primitive_msg)
+{
+    validatePrimitiveMessage(primitive_msg, getPrimitiveName());
+
+    robot_id     = primitive_msg.robot_id;
+    wheel0_power = int(primitive_msg.parameters.at(0));
+    wheel1_power = int(primitive_msg.parameters.at(1));
+    wheel2_power = int(primitive_msg.parameters.at(2));
+    wheel3_power = int(primitive_msg.parameters.at(3));
+    dribbler_rpm = primitive_msg.parameters.at(4);
```

Comment:
I think we should throw an exception here if any of the wheel powers are out of bounds?

Reply:
I think this is a perfect place to clamp the value and log a warning instead? Or a combination of both?